WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderTagAdapter.java

```diff
@@ -99,36 +102,48 @@ public ReaderTagList getItems() {
         return mTags;
     }
 
+    @Nullable
+    public ReaderTagList getSubscribedItems() {
+        final ReaderTagList readerSubscribedTagsList = new ReaderTagList();
+        for (final ReaderTag readerTag : mTags) {
+            if (Boolean.TRUE.equals(mBlogIdIsFollowedMap.get(readerTag.getTagSlug()))) {
+                readerSubscribedTagsList.add(readerTag);
+            }
+        }
+        return readerSubscribedTagsList;
+    }
+
     @Override
     public void onBindViewHolder(TagViewHolder holder, int position) {
         final ReaderTag tag = mTags.get(position);
         holder.mTxtTagName.setText(tag.getLabel());
-        holder.mRemoveFollowButton.setOnClickListener(v -> performDeleteTag(tag));
+        holder.mRemoveFollowButton.setOnClickListener(view -> performDeleteTag(tag, holder.mRemoveFollowButton));
     }
 
-    private void performDeleteTag(@NonNull ReaderTag tag) {
+    private void performDeleteTag(@NonNull ReaderTag tag, @NonNull final ReaderFollowButton readerFollowButton) {
         if (!NetworkUtils.checkConnection(getContext())) {
             return;
         }
 
-        ReaderActions.ActionListener actionListener = new ReaderActions.ActionListener() {
-            @Override
-            public void onActionResult(boolean succeeded) {
-                if (!succeeded && hasContext()) {
-                    ToastUtils.showToast(getContext(), R.string.reader_toast_err_removing_tag);
-                    refresh();
-                }
+        final boolean currentFollowValue = Boolean.TRUE.equals(mBlogIdIsFollowedMap.get(tag.getTagSlug()));
+        final boolean newFollowValue = !currentFollowValue;
+        readerFollowButton.setIsFollowed(newFollowValue);
+
+        // Disable follow button until API call returns
+        readerFollowButton.setEnabled(false);
+
+        ReaderActions.ActionListener actionListener = succeeded -> {
+            mBlogIdIsFollowedMap.put(tag.getTagSlug(), newFollowValue);
+            readerFollowButton.setEnabled(true);
+            if (!succeeded && hasContext()) {
+                ToastUtils.showToast(getContext(), R.string.reader_toast_err_removing_tag);
+                refresh();
             }
         };
 
         boolean success = ReaderTagActions.deleteTag(tag, actionListener, mAccountStore.hasAccessToken());
```

Comment:
🚫 It looks like we are only ever deleting the tags, even if `newFollowValue` is `true`. Shouldn't we add back the tag if the user tapped the button to unsubscribe but then decided to keep subscribed?

Right now, the behavior is that even tapping the "Subscribe" button that appears the tag is still deleted.

Reply:
Thanks for catching that! It's fixed and also made me realize I wasn't calling the `reader_reader_tag_followed` analytics event when the tag is being subscribed from the follow button. This is also fixed.